### PR TITLE
Gauss Newton with Line Search

### PR DESCRIPTION
**Pull request by avik-pal**
# TODOs

- [ ] Fix Tests
- [x] Wait on https://github.com/SciML/NonlinearSolve.jl/pull/265
- [x] Upstream patch to allow non-square function operators for VecJac https://github.com/JuliaDiff/SparseDiffTools.jl/pull/272
- [x] Remove Manifest

### CODE REVIEW

**Comment by vpuri3** [comment]
Any reason to not use `LineSearches.jl`?

**Comment by avik-pal** [comment]
It is using LineSearches internally

**Comment by avik-pal** [comment]
This is done. just need the previous PRs to be merged

File: Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,954 @@
+# This file is machine-generated - editing it directly is not advised
+
+julia_version = "1.10.0-beta3"
```
Comment (ChrisRackauckas) [inline Manifest.toml:3]: remove?

**Comment by vpuri3** [comment]
would it be possible to pass an initial guess for alpha and line search? example? InitialStatic(), InitialPrevious(), etc...

File: Manifest.toml (hunk 1 above)
Comment (avik-pal) [reply Manifest.toml:3]: it is to run the CI with the SparseDiffTools PR

**Comment by avik-pal** [comment]
> would it be possible to pass an initial guess for alpha and line search? example? InitialStatic(), InitialPrevious(), etc...

Not currently. I have no clue how that API currently works.

**Pull request merged**